First version of Horizontal component

Flemingjp opened the pull request:
Completed the first version of the Horizontal component. This includes

- Toggle between radio list and drop down menu
- Logging of selected answer
- Styling for text
- Styling for sizing
- Styling for alignment
- Working story for the component

Commit:
Set up loading in options to components and added relevant stories with data input to test compoennt

Commit:
Added working DropDownList and RadioButtonList component - logResults on option selection

Commit:
Update to package.json

Flemingjp committed:
Added sub component RadioListItem to simplify styling. Adding inital styling to RadioButtonList

Flemingjp committed:
Moved styles to top level component

Flemingjp committed:
Removed icon and fixed styles access in RadioListItem to allow Decsys to load properly

Flemingjp committed:
Added explicit styling properties to parent Component, and completed styling coverage in Storybook and Decsys.

Flemingjp committed:
Changed option=>lable in option object to support React-Select better. Adding styling to dropdownlist which now using React-Select

Flemingjp committed:
Added alignment and width options to component

Flemingjp committed:
Removed reload command from package.json
Removed FlexBox.js component
Renaming ListContainer to FlexBox for consistency with other components @decsys

Flemingjp committed:
Added Icon for Component
Added default option for list items

Flemingjp committed:
Removed unused imports

beforan commented on package.json:
```diff
@@ -1,9 +1,9 @@
-  "name": "my-component",
-  "componentName": "MyComponent",
+  "name": "@decsys/horizontal-component",
+  "componentName": "Horizontal",
```
We're gonna want a better name ;)

spaces are acceptable.

I think I suggested something like "Multiple Choice" earlier? But open to suggestions.

"Select" or "One Of" or something might also work. But Horizontal has nothing too do with the kind of response they give, or the means of giving.

Purely a description of its layout which is probably also wrong sometimes (e.g. for a dropdown box, or a vertical list o radios?)

beforan commented on src/Component.js:
```diff
@@ -1,27 +1,52 @@
+// Container to sort out component alignment and size
+const Alignment = styled.div`
+  width: ${x => x.dropDown ? (x.width || "100%") : "fit-content"};
+  float: ${x => (x.alignment == "center") ? "" : x.alignment};
```
any chance we can use a `display: flex` rather than floats / margins? `float` is basically awful and rarely needed since flex/grid.

beforan commented on src/Component.js:
```diff
@@ -1,27 +1,52 @@
+  }
 
   return (
+    // Parent <div> required for alignment in Decsys
```
display flex on Alignment might also resolve the need for this wrapper. but don't quote me on that.

beforan commented on src/components/RadioButtonList.js:
```diff
@@ -0,0 +1,26 @@
+        <FlexBox>
+        {
+            options.map(option =>
+                <RadioListItem name={name} option={option} onSelection={onSelection} {...props}/>    
```
React will want you to put a `key` prop on this. https://reactjs.org/docs/lists-and-keys.html

the `name` will do fine I think.

Flemingjp committed:
Added FlexBox component to replace Alignmnet component -> uses flex rather than floats

Flemingjp committed:
Added key parameter to radio list
Renamed logOption=>onSelection for DropDownList to be consistent with RadioButtonList
Renamed RadioButtonList to RadioList for more consistent naming

Flemingjp committed:
Moved key from RadioList to RadioListItem
Updated Component imports

Flemingjp committed:
Renamed component to Choose One to reflect its functionality better

beforan commented on src/components/RadioListItem.js:
```diff
@@ -0,0 +1,26 @@
+
+const RadioListItem = ({ name, option, onSelection, ...props }) => (
+    <StyledLabel {...props}>
+        <StyledRadio key={name} name={name} onChange={() => onSelection(option)} {...props} />
```
`key` needs moving, now you've refactored 😄 it needs to be on the imediate child of the `.map()` call, essentially

beforan commented on src/components/RadioList.js:
```diff
@@ -0,0 +1,21 @@
+        <FlexBox>
+        {
+            options.map(option =>
+                <RadioListItem name={name} option={option} onSelection={onSelection} {...props}/>    
```
`key` needs to be here now

Flemingjp committed:
Moved key to RadioListItem
